noria-server/src/controller/sql/mir/mod.rs

```diff
@@ -1056,6 +1124,34 @@ impl SqlToMirConverter {
                 vec![parent_node.clone()],
                 vec![],
             ),
+            GroupedNodeType::FilterAggregation(filter_agg) => {
+                use nom_sql::ConditionExpression::*;
+
+                let cond = condition.expect("Filter aggregation must have condition!");
```

Comment:
Given that this is the case, it might be better to just stick the conditions inside the `FIlterAggregation` variant rather than passing it separately?

Reply:
I spent a while trying to do this, and it's actually a huge pain, primarily because mir_to_flow also wants to pass conditions along with the GroupedNodeType but in that case the conditions are Vec<(usize, FilterCondition)>s rather than ConditionExpressions. So then I tried just making them both Vec<(usize, FilterCondition)>s and just pushing the logic to convert from ConditionExpressions earlier in mir/mod.rs. But this snowballed into pushing a lot of logic earlier because to_conditions requires already having made the parent node...

I'm going to give up on this one, though I agree it would've been a lot more elegant.